AssertionError in erasure when overriding generic trait method with erased using clause before type-param argument

## Compiler version

3.8.3 (also verified on 3.5.2, 3.6.4, 3.7.2, 3.7.3, 3.8.0, 3.8.1)

## Minimized code

```scala
import scala.language.experimental.erasedDefinitions

class Ev
trait Svc[A]:
  def run(using erased Ev)(a: A): String
class ISvc extends Svc[Int]:
  def run(using erased Ev)(a: Int): String = a.toString
```

## Output (click arrow to expand)

<details>

```
  unhandled exception while running erasure on bug004.scala

  An unhandled exception was thrown in the compiler.
  Please file a crash report here:
  https://github.com/scala/scala3/issues/new/choose

     while compiling: bug004.scala
        during phase: erasure
                mode: Mode(ImplicitsEnabled)
     library version: version 3.8.3
    compiler version: version 3.8.3

Exception in thread "main" java.lang.AssertionError: assertion failed: bad adapt for this.run(): (x$0: Int): String
	at scala.runtime.Scala3RunTime$.assertFailed(Scala3RunTime.scala:10)
	at dotty.tools.dotc.transform.Erasure$Boxing$.adaptToType(Erasure.scala:389)
	at dotty.tools.dotc.transform.Erasure$Typer.adapt(Erasure.scala:1062)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3984)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3989)
	at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:4101)
	at dotty.tools.dotc.typer.Typer.$anonfun$69(Typer.scala:3141)
	at dotty.tools.dotc.inlines.PrepareInlineable$.dropInlineIfError(PrepareInlineable.scala:256)
	at dotty.tools.dotc.typer.Typer.typedDefDef(Typer.scala:3141)
	at dotty.tools.dotc.transform.Erasure$Typer.typedDefDef(Erasure.scala:936)
	at dotty.tools.dotc.typer.Typer.typedNamed$1(Typer.scala:3785)
	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3896)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3984)
```

</details>

## Expectation

Should compile successfully.

## Key observation: requires generic type parameter + erased BEFORE type-param arg

The non-generic version compiles fine:
```scala
// OK — no type parameter
trait Svc:
  def run(using erased Ev)(a: Int): String
class ISvc extends Svc:
  def run(using erased Ev)(a: Int): String = a.toString
```

ICE only when the trait has a type parameter `[A]` and the method uses `(a: A)`, with erased parameter coming BEFORE the type-param-derived argument:
```scala
(using erased Ev)(a: A)           // ICE — erased before type-param arg
(erased ev: Ev)(a: A)             // ICE — same, without `using`
(erased ev: Ev, a: A)             // ICE — same param list
```

OK when erased comes AFTER:
```scala
(a: A)(using erased Ev)           // OK — erased after type-param arg
```

## Impact: also affects saferExceptions

`CanThrow` from `scala.language.experimental.saferExceptions` is an erased class. This pattern also crashes:
```scala
import language.experimental.saferExceptions

class MyEx extends Exception
trait Processor[A]:
  def process(using CanThrow[MyEx])(a: A): String
class IntProcessor extends Processor[Int]:
  def process(using CanThrow[MyEx])(a: Int): String = a.toString  // ICE!
```

## Notes
- Long-standing: verified on every version from 3.5.2 through 3.8.3.
- Workaround: put the erased using clause AFTER the type-param-derived argument.
- Found using my Scala 3 grammar model and [DepTyCheck](https://github.com/buzden/deptycheck)
Also found via my tools: https://github.com/scala/scala3/issues/25724, https://github.com/scala/scala3/issues/25723, https://github.com/scala/scala3/issues/25248, https://github.com/scala/scala3/issues/25246, https://github.com/scala/scala3/issues/25244

--- a/compiler/src/dotty/tools/dotc/transform/Erasure.scala
+++ b/compiler/src/dotty/tools/dotc/transform/Erasure.scala
@@ -804,8 +804,9 @@ object Erasure {
       val Apply(fun, args) = tree
       val origFun = fun.asInstanceOf[tpd.Tree]
       val origFunType = origFun.tpe.widen(using preErasureCtx)
+      val insideBridge = ctx.owner.ownersIterator.exists(_.is(Flags.Bridge))
       val ownArgs = origFunType match
-        case mt: MethodType if mt.hasErasedParams =>
+        case mt: MethodType if mt.hasErasedParams && !insideBridge =>
           args.lazyZip(mt.paramErasureStatuses).flatMap: (arg, isErased) =>
             if isErased then
               checkPureErased(arg, isArgument = true,

--- a/tests/pos/i25726.scala
+++ b/tests/pos/i25726.scala
@@ -1,0 +1,48 @@
+import scala.language.experimental.erasedDefinitions
+
+class Ev
+
+trait Svc[A]:
+  def run(using erased Ev)(a: A): String
+class ISvc extends Svc[Int]:
+  def run(using erased Ev)(a: Int): String = a.toString
+
+// More than one pre-erasure parameter list with an all-erased prefix
+class Token
+trait Store[A]:
+  def put(using erased Token)(key: String)(value: A): String
+class IntStore extends Store[Int]:
+  def put(using erased Token)(key: String)(value: Int): String = s"$key=$value"
+
+// Trailing all-erased list (worked before the fix; must not regress)
+trait Trail[A]:
+  def id(a: A)(using erased Ev): A
+class IntTrail extends Trail[Int]:
+  def id(a: Int)(using erased Ev): Int = a
+
+// Erased-first inside a single parameter list
+trait Single[A]:
+  def run(erased a: Ev, b: A): String
+class IntSingle extends Single[Int]:
+  def run(erased a: Ev, b: Int): String = b.toString
+
+// Erased-first within a non-trailing parameter list
+trait Mixed[A]:
+  def run(erased e: Ev, a: A)(b: Int): String
+class IntMixed extends Mixed[Int]:
+  def run(erased e: Ev, a: Int)(b: Int): String = s"$a+$b"
+
+  trait Eta[A]:
+    def f(using erased Ev)(a: A): String ?=> Int
+
+  class IntEta extends Eta[Int]:
+    def f(using erased Ev)(a: Int): String ?=> Int = a + summon[String].length
+
+// generated bridge:
+// def f(a: Object): Function1[String, Int] = // beridge
+//   new Function1[String, Int]:
+//     def apply(s: String): Int = this$IntEta.f(a, s)
+trait Closure[A]:
+  def f(using erased Ev)(a: A): String ?=> Int
+class IntClosure extends Closure[Int]:
+  def f(using erased Ev)(a: Int): String ?=> Int = a + summon[String].length

